src/token.rs

```diff
@@ -73,20 +73,28 @@ impl<'a> Token<'a> {
         self.kind == TokenKind::StopWord
     }
 
-    /// Returns the number of graphemes in original token using number of characters in normalized
+    /// Returns the number of chars in original token using number of bytes in normalized
     /// token.
     ///
-    /// Grapheme clusters are counted in the pre-processed string (just before normalizing).
+    /// chars are counted in the pre-processed string (just before normalizing).
     /// For example, consider the string "Go💼od" which gets normalized to "gobriefcase od".
-    /// `num_graphemes_from_bytes(11)` for this token will return `3` - the number of characters in
+    /// `num_chars_from_bytes(11)` for this token will return `3` - the number of characters in
     /// the original string for 11 bytes in the normalized string.
     ///
     /// # Arguments
     ///
     /// * `num_bytes` - number of bytes in normalized token
-    pub fn num_graphemes_from_bytes(&self, mut num_bytes: usize) -> usize {
+    pub fn num_chars_from_bytes(&self, mut num_bytes: usize) -> usize {
         match &self.char_map {
-            None => self.word.len(),
+            None => {
+                // if we don't have a char_map, we look for the number of chars in the current
+                //   (probably normalized) string
+                self.word
+                    .char_indices()
+                    .take_while(|(byte_index, _)| *byte_index <= num_bytes)
+                    .count()
+                    - 1
```

Comment:
why `-1`?
if num_bytes = 1 then count may be 1 and then we return 0

Reply:
Good catch!

Although, it won't fail when `num_bytes` is 1. The char indices look like this (for "word" string): `[0, 1, 2, 3]`. So the count will be `2` for `num_bytes = 1`.

It will fail for `num_bytes` = `word.len()`. In the above, when `num_bytes` is 4, the function returns 3 which is not right.

Reply:
I have fixed this in the latest commit!